test/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.test.cpp

```diff
@@ -355,25 +355,54 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Dynamics_Tabulated, ComputeContribution)
             defaultInterpolationType_,
         };
 
-        // Wrong frame
+        // Same frame as contribution profile
         {
-            EXPECT_THROW(
-                tabulated.computeContribution(defaultInstants_[0], x, Frame::ITRF()), ostk::core::error::runtime::Wrong
+            EXPECT_EQ(
+                tabulated.computeContribution(defaultInstants_[0], x, defaultFrameSPtr_), contributionProfile_.row(0)
+            );
+        }
+
+        // Different frame as contribution profile (contributions stored in quasi-inertial frame, but requested in
+        // non-quasi inertial)
+        {
+            EXPECT_NE(
+                tabulated.computeContribution(defaultInstants_[0], x, Frame::ITRF()), contributionProfile_.row(0)
             );
         }
 
+        // Different frame as contribution profile (contributions stored in non-quasi-inertial frame, but requested in
+        // quasi inertial, so expect a warning logged)
+        {
+            Tabulated tabulatedNonQuasiInertialFrame = {
+                defaultInstants_,
+                contributionProfile_,
+                defaultWriteCoordinateSubsets_,
+                Frame::ITRF(),
+                defaultInterpolationType_,
+            };
+
+            testing::internal::CaptureStdout();
+
+            const VectorXd computedContribution =
+                tabulatedNonQuasiInertialFrame.computeContribution(defaultInstants_[0], x, defaultFrameSPtr_);
+
+            EXPECT_FALSE(testing::internal::GetCapturedStdout().empty());
+
+            EXPECT_NE(computedContribution, contributionProfile_.row(0));
```

Comment:
Instead of a NE check, we should do an Eq check with the correct expected rotation.